File: fib.py
def fib(index, seq=None):
    if seq == None:
        seq = [0, 1]
    if index == 1:
        return seq[-1]

    seq.append(seq[-2] + seq[-1])
    index = index - 1
    return fib(index, seq)

print(fib(7))

File: fib.ex
defmodule Fib do
  def fib(index, seq \\ [0, 1])
  def fib(1, seq) do
    Enum.at(seq, -1)
  end
  def fib(index, seq) do
    tail = Enum.at(seq, -1) + Enum.at(seq, -2)
    fib(index - 1, seq ++ [tail])
  end
end

IO.inspect Fib.fib(7)
